Lib/os.py

```diff
@@ -1069,7 +1069,9 @@ def __fspath__(self):
 
     @classmethod
     def __subclasshook__(cls, subclass):
-        return hasattr(subclass, '__fspath__')
+        if cls is PathLike and hasattr(subclass, '__fspath__'):
+            return True
+        return NotImplemented
```

Comment:
Is it supposed to have any special behaviour if `__fspath__` is `None`? I am just curious if it is possible to unify this with behaviour of `collections.abc` classes, and use the `_check_methods()` helper, see e.g. [`Hashable`](https://github.com/python/cpython/blob/master/Lib/_collections_abc.py#L95).

Reply:
I'll be honest with you, I'm not entirely sure why it's not in collections.abc anyway.
You might say it fits in os because it's not "python-wide", but then again you can find coroutine stuff in collections, outside asyncio.
`_check_methods()` is internal, which is why I did not use it. Didn't wish to re-implement it outside. The only difference is `None` handling, which doesn't seem as important (and other dynamic `__getattr__` stuffs).
`Hashable` is a difference scenario - classes come with a built-in `__hash__` that needs to be explicitly silenced. This is not the case.

Reply:
@ilevkivskyi it would be an error to set the method to `None` since that's the key point of the protocol.

@bharel you can read the python-dev archives on PEP 519 if you want, but there are reasons (if I remember correctly the decision was made to put ABCs closer to the code that it relates to rather than centralize in `collections.abc`; plus this isn't a collection).

And this is the stdlib, you can always use private stuff if you're careful about it.

Reply:
> `None` provides no special meaning.

OK. Note however that protocols always treat `None` as special (both user-defined PEP 544, and the "builtin" ones from `collections.abc`). For example:
```python
from collections.abc import Iterable
from typing import Protocol, runtime_checkable

class NotReallyIterable:
    __iter__ = None

isinstance(NotReallyIterable(), Iterable)  # returns False

@runtime_checkable
class P(Protocol):
    def whatever(self) -> None: ...

class Yes:
    def whatever(self) -> None: ...
class No:
    whatever = None

isinstance(No(), P)  # False
isinstance(Yes(), P)  # True
```
Anyway, this is probably not very important. Just wanted to point out this corner case.

Reply:
@brettcannon afaik an awaitable isn't a collection either, should be in `asyncio.abc`. Origin came from generators but the meaning is miles apart.
Nor a Hashable. Unless every custom class in python is a collection, I find that argument invalid.

Reply:
@bharel you can find it invalid if you choose, but we had our reasons and I do not want to get into an argument over it, especially as it is not consequential to this PR.

Reply:
@brettcannon sure, all good :-)
If I can use the private function _check_methods, I'll import it and use it instead of hasattr. The code will match the one in collections.abc. Thanks Brett!